From 0ccada26696ab7f1f49f29f13c7d0109e8da8f94 Mon Sep 17 00:00:00 2001
From: Yong Zhi <yong.zhi@intel.com>
Date: Tue, 27 Feb 2018 10:44:14 -0600
Subject: [PATCH 1891/2313] media: intel-ipu4: isys: v4l2-compliance fix for
 CSI-2 node

Message for Open Source:
Fix v4l2-compliance fails on "Intel IPU4 CSI-2 2 capture" node.

Message for Internal:
[Issue/Feature] v4l2-compliance -d /dev/video4 returns:

"fail: pixelformat 6d347069 (ip4m) not reported by ENUM_FMT"
"test VIDIOC_G_FMT: FAIL"

For /dev/video4, set pixelformat to 0 causes
"Unknown pixelformat ..." kernel warning.

[Root Cause/Changes] Missing implementation.

Change-Id: Ic5b0abef663424f06dd1ab4e92493421e45378cd
Signed-off-by: Yong Zhi <yong.zhi@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-isys-video.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys-video.c b/drivers/media/pci/intel/ipu-isys-video.c
index c709e4a..a3af92c 100644
--- a/drivers/media/pci/intel/ipu-isys-video.c
+++ b/drivers/media/pci/intel/ipu-isys-video.c
@@ -414,12 +414,6 @@ int ipu_isys_vidioc_enum_fmt(struct file *file, void *fh,
 
 	f->flags = 0;
 
-	if (*supported_codes == MEDIA_BUS_FMT_FIXED) {
-		/* FIXME: No proper pixel format. Use 0. */
-		f->pixelformat = 0;
-		return 0;
-	}
-
 	/* Code found */
 	for (pfmt = av->pfmts; pfmt->bpp; pfmt++)
 		if (pfmt->code == *supported_codes)
@@ -570,6 +564,10 @@ static void fmt_mp_to_sp(struct v4l2_pix_format *pix,
 	pix->bytesperline = mpix->plane_fmt[0].bytesperline;
 	pix->sizeimage = mpix->plane_fmt[0].sizeimage;
 	pix->flags = mpix->flags;
+	pix->colorspace = mpix->colorspace;
+	pix->ycbcr_enc = mpix->ycbcr_enc;
+	pix->quantization = mpix->quantization;
+	pix->xfer_func = mpix->xfer_func;
 }
 
 static int vidioc_g_fmt_vid_cap(struct file *file, void *fh,
-- 
2.7.4

